addons/sale/tests/test_access_rights.py

```diff
@@ -0,0 +1,135 @@
+            'user_id': self.user_salesperson.id
+        })
+
+    def test_access_sales_manager(self):
+        """ Test sales manager's access rights """
+        # Manager can see the SO of other salesperson
+        self.order.sudo(self.user_manager).read
+        # Manager can create the SO of other salesperson
+        sale_order = self.env['sale.order'].sudo(self.user_manager).create({
+            'partner_id': self.partner_customer_usd.id,
+            'user_id': self.user_salesperson_1.id
+        })
+        self.assertIn(sale_order.id, self.env['sale.order'].search([]).ids, 'Sales manager should be able to create the SO of other salesperson')
+
+        # Manager can change a salesperson of the SO
+        with Form(sale_order) as order:
+            order.user_id = self.user_salesperson
+        # Manager can delete the SO of other salesperson
+        sale_order.unlink()
+        self.assertNotIn(sale_order.id, self.env['sale.order'].search([]).ids, 'Sales manager should be able to delete the SO')
+
+        # Manager can create a sales channel
+        india_channel = self.env['crm.team'].sudo(self.user_manager).create({
+            'name': 'India',
+        })
+        self.assertIn(india_channel.id, self.env['crm.team'].search([]).ids, 'Sales manager should be able to create a sales channel')
+        # Manager can edit a sales channel
+        with Form(india_channel) as team:
+            team.dashboard_graph_group = 'week'
+        self.assertEquals(india_channel.dashboard_graph_group, 'week', 'Sales manager should be able to edit a sales channel')
+        # Manager can delete a sales channel
+        india_channel.unlink()
+        self.assertNotIn(india_channel.id, self.env['crm.team'].search([]).ids, 'Sales manager should be able to delete a sales channel')
+
+    def test_access_sales_person(self):
+        """ Test Salesperson's access rights """
+        # Salesperson can see only his/her sale order
+        self.assertRaises(AccessError, self.order.sudo(self.user_salesperson_1).read, ['user_id'])
+
+        # Salesperson cann't create the SO of other salesperson
+        with self.assertRaises(AccessError):
+            self.env['sale.order'].sudo(self.user_salesperson_1).create({
+                'partner_id': self.partner_customer_usd.id,
+                'user_id': self.user_salesperson.id
+            })
+
+        # Now assign SO to himself/herself
+        sale_order_1 = self.env['sale.order'].sudo(self.user_salesperson_1).create({
+            'partner_id': self.partner_customer_usd.id,
+        })
+        # Salesperson cann't delete the SO
+        with self.assertRaises(AccessError):
+            sale_order_1.unlink()
+
+        # Salesperson can see a sales channel
+        sale_order_1.team_id.read
+        # Salesperson can change a sales channel of SO
+        with Form(sale_order_1) as order:
+            order.team_id = self.env.ref('sales_team.crm_team_1')
```

Comment:
Don't use demo data in test. Either you create one in the `SetUp`method, either you modifier another field on SO form view.